eng/Packages.props

```diff
@@ -23,13 +23,26 @@
     <PackageVersion Include="System.Net.Http" Version="$(SystemNetHttpVersion)" />
     <PackageVersion Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" />
     <PackageVersion Include="System.Reflection.MetadataLoadContext" Version="$(SystemReflectionMetadataLoadContextVersion)" />
-    <PackageVersion Include="System.Resources.Extensions" Version="$(SystemResourcesExtensionsPackageVersion)" />
+    <PackageVersion Include="System.Resources.Extensions" Version="$(SystemResourcesExtensionsVersion)" />
     <PackageVersion Include="System.Runtime.CompilerServices.Unsafe" Version="$(SystemRuntimeCompilerServicesUnsafeVersion)" />
     <PackageVersion Include="System.Security.Principal.Windows" Version="$(SystemSecurityPrincipalWindowsVersion)" />
     <PackageVersion Include="System.Text.Encoding.CodePages" Version="$(SystemTextEncodingCodePagesVersion)" />
     <PackageVersion Include="System.Text.Json" Version="$(SystemTextJsonVersion)" />
     <PackageVersion Include="System.Threading.Channels" Version="$(SystemThreadingChannelsVersion)" />
     <PackageVersion Include="System.Threading.Tasks.Dataflow" Version="$(SystemThreadingTasksDataflowVersion)" />
     <PackageVersion Include="xunit.console" Version="$(XUnitVersion)" />
+
+  </ItemGroup>
+
+  <!-- This set of packages will use different versions based on building binaries for VS. In non-UB modes this has no effect. -->
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
```

Comment:
The logical expression `TFI == .NETFramework => Building binaries for VS` isn't true. It's very important to keep that in mind. I just last week had the same discussion on this with tmat and we landed on a nice design in Arcade: https://github.com/dotnet/arcade/pull/14391#discussion_r1465461640

Reply:
VS/packages really. IMO the key bit here is that is really the version flow problem I mentioned yesterday. We should not be transparently upgrading these dependencies except in source-only modes.

Reply:
@ViktorHofer I don't think I understand the concern. Can you elaborate?

Reply:
The changes here have since been reverted. in non source-only modes, msbuild builds very very close to how it builds in the isolated repo context.

Reply:
> @ViktorHofer I don't think I understand the concern. Can you elaborate?

Pinning package versions globally to a specific version for VS support based on the `$(TargetFramework` or `$(TargetFrameworkIdentifier)` is an incorrect assumption. You could have a normal class library that uses i.e. System.Text.Json and multi-targets for both .NET Framework and .NETCoreApp. Pinning the System.Text.Json version down to X for the .NET Framework build doesn't make sense. VS support applies to msbuild tasks, etc but not to class libraries that ship out-of-band via nuget packages.

My general recommendation around this has been to mark projects that apply to these restrictions via some property, i.e. https://github.com/dotnet/arcade/blob/2d7ad177f759e8a0128cc53fa90ffc54aec1822c/src/Microsoft.DotNet.NuGetRepack/tasks/Microsoft.DotNet.NuGetRepack.Tasks.csproj#L6.

In the example of Arcade, it then imports a targets file that updates the PackageVersion items for VS support: https://github.com/dotnet/arcade/blob/2d7ad177f759e8a0128cc53fa90ffc54aec1822c/eng/BuildTask.targets#L3

Reply:
> VS support applies to msbuild tasks, etc but not to class libraries that ship independently.

I think I agree with this but I don't see how it applies to this repo, which is filled with things where we only care about the .NET Framework versions of things because of VS specifically.